### ISSUE TITLE
Proper unit testing runner.

### ISSUE THREAD

**Issue by MattWindsor91** [septract/starling-tool#5]
In the (as of writing not yet merged) tool, one passes `-t` to switch to unit testing mode: Fuchu then expects to be given total control of the argv of the program so it can parse its own arguments.  Currently we do this, but only after `CommandLine` has parsed them (which means in practice Fuchu can only have one argument passed to it, because Starling usually only permits one non-dashed argument through `CommandLine`).

This is a bit messy, so either we need to:
- move to something with a proper testing runner (NUnit?);
- figure out how to build a separate test binary;
- fix the command line (maybe make it so we check if the first argument is `test` before we use the command parser?).

**Comment by MattWindsor91** [comment]
We now use nunit, so this is no longer relevant; closing.
